modules/configuration/src/main/java/org/apache/ignite/internal/configuration/validation/CamelCaseKeysValidator.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.configuration.validation;
+
+import java.util.regex.Pattern;
+import org.apache.ignite.configuration.NamedListView;
+import org.apache.ignite.configuration.validation.CamelCaseKeys;
+import org.apache.ignite.configuration.validation.ValidationContext;
+import org.apache.ignite.configuration.validation.ValidationIssue;
+import org.apache.ignite.configuration.validation.Validator;
+
+/** {@link Validator} implementation for the {@link CamelCaseKeys} annotation. */
+public class CamelCaseKeysValidator implements Validator<CamelCaseKeys, NamedListView<?>> {
+    private static final String CAMEL_CASE_GOOGLE_STYLE = "https://google.github.io/styleguide/javaguide.html#s5.3-camel-case";
```

Comment:
Why have you decide to use Google's regex, did you check this pattern yourself?
It looks really weird, and it contradicts my intuition. Especially the last part - `([A-Z])?`, why does it exist?
`(\\d)|([A-Z0-9]...` is also weird, it means that we allow both "a5b" and "a5B", why would we do it :)
Why do we allow digits in the first place? I guess this guide is inappropriate for the task.
I recommend re-writing it manually so that it fits our goals, and also remove the reference to Google guidelines. What do you think?

Reply:
We discussed it in private messages.